### ISSUE TITLE
Ability to flush logs

### ISSUE THREAD

**Issue by taythebot** [ayZagen/pino-kafka#1]
Is there a way to flush the logs? It's important to be able to flush logs before the process exits in case it crashes. 

When using it with the `pino.final` method it gives this error `Error: final requires a stream that has a flushSync method, such as pino.destination`

Btw thank you so much for the library. This is the only Pino Kafka library I could find.

**Comment by ayZagen** [comment]
Because of `node-rdkafka` methods are being asynchronous I won't implement the interface required for `pino.final` because it won't work as it intent to be.

But what we could do is, I could expose kafka client to pino stream and with trapping process signals we can handle kafka flushing. 
Here is example code:
```js
function terminator( sig ) {
  if ( typeof sig === 'string' ) {
    flushKafkaQueue( () => {
      process.exit( 1 );
    });
  }
}

async function flushKafkaQueue( callback ){
 logger[pino.symbols.streamSym]._kafka.flush(1000, err => {
   if(err){
     // do something with internal kafka err
   }
   callback()
 })
}

['SIGHUP', 'SIGINT', 'SIGQUIT', 'SIGILL', 'SIGTRAP', 'SIGABRT',
  'SIGBUS', 'SIGFPE', 'SIGUSR1', 'SIGSEGV', 'SIGUSR2', 'SIGTERM'
].forEach( ( sig ) => {
  process.on( sig, () => {
    try {
      terminator( sig );
    } catch ( e ) {
      process.exit(1)
    }
  } );
} );
```
Also I believe this should be on user land and not in library because trapping signals from library could lead unexpected behavior for user app.
I will publish new version with kafka stream exposed in `_kafka`.

**Comment by ayZagen** [comment]
Published 0.1.1
[related readme section](../#accessing-internal-kafka-producer)
